src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisInput.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.sensitivityanalysis.server.service;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.contingency.Contingency;
+import com.powsybl.contingency.ContingencyContext;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.IdentifiableType;
+import com.powsybl.iidm.network.Load;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.sensitivity.SensitivityFactor;
+import com.powsybl.sensitivity.SensitivityFunctionType;
+import com.powsybl.sensitivity.SensitivityVariableSet;
+import com.powsybl.sensitivity.SensitivityVariableType;
+import com.powsybl.sensitivity.WeightedSensitivityVariable;
+import org.gridsuite.sensitivityanalysis.server.dto.IdentifiableAttributes;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+public class SensitivityAnalysisInput {
+    private final Network network;
+    private final SensitivityAnalysisRunContext context;
+    private final ActionsService actionsService;
+    private final FilterService filterService;
+
+    private List<Contingency> contingencies = new ArrayList<>();
+    private List<SensitivityFactor> factors = new ArrayList<>();
+    private List<SensitivityVariableSet> variableSets = new ArrayList<>();
+    private List<IdentifiableAttributes> identifiables = new ArrayList<>();
+
+    public SensitivityAnalysisInput(Network network, SensitivityAnalysisRunContext context, ActionsService actionsService, FilterService filterService) {
+        this.network = network;
+        this.context = context;
+        this.actionsService = actionsService;
+        this.filterService = filterService;
+    }
+
+    private void buildContingencies() {
+        contingencies = context.getContingencyListUuids().stream()
+            .flatMap(contingencyListUuid -> actionsService.getContingencyList(contingencyListUuid, context.getNetworkUuid(), context.getVariantId()).stream())
+            .collect(Collectors.toList());
+    }
+
+    private void buildSensitivityVariableSets() {
+        List<List<IdentifiableAttributes>> variablesFiltersLists = context.getVariablesFiltersListUuids().stream()
+            .map(variablesFilterUuid -> filterService.getIdentifiablesFromFilter(variablesFilterUuid, context.getNetworkUuid(), context.getVariantId()))
+            .collect(Collectors.toList());
+
+        variablesFiltersLists.forEach(variablesList -> {
+            List<WeightedSensitivityVariable> variables = new ArrayList<>();
+            if (!variablesList.isEmpty()) {
```

Comment:
The sum is not needed, a weight is expected.This is not necessary to nomalize.

Reply:
Done